node/core/backing/src/lib.rs

```diff
@@ -1136,22 +1153,8 @@ async fn seconding_sanity_check<Context>(
 	candidate_hash: CandidateHash,
 	candidate_para: ParaId,
 	parent_head_data_hash: Hash,
-	head_data_hash: Hash,
 	candidate_relay_parent: Hash,
 ) -> SecondingAllowed {
-	// Note that `GetHypotheticalDepths` doesn't account for recursion,
-	// i.e. candidates can appear at multiple depths in the tree and in fact
-	// at all depths, and we don't know what depths a candidate will ultimately occupy
-	// because that's dependent on other candidates we haven't yet received.
-	//
-	// The only way to effectively rule this out is to have candidate receipts
-	// directly commit to the parachain block number or some other incrementing
-	// counter. That requires a major primitives format upgrade, so for now
-	// we just rule out trivial cycles.
-	if parent_head_data_hash == head_data_hash {
```

Comment:
Any comment on the removal of this check? It easily eliminates most types of cycles

Reply:
(I see it was moved elsewhere)